### PR TITLE
perf(collaboration): update collaboration return type

### PR DESCRIPTION
**Pull request by SLdragon**

### CODE REVIEW

File: packages/fx-core/src/common/permissionInterface.ts [hunk 1]
```diff
@@ -8,6 +8,30 @@ export interface ResourcePermission {
   roles: string[] | undefined;
 }
 
+export enum CollaborationState {
+  OK,
+  NotProvisioned,
+  M365AccountNotMatch,
```
Comment (KennethBWSong) [inline packages/fx-core/src/common/permissionInterface.ts:14]: Not match seems confusing. What do you mean by not match here?

Comment (SLdragon) [reply packages/fx-core/src/common/permissionInterface.ts:14]: User login a different m365 account that is not belong to current project

Comment (KennethBWSong) [reply packages/fx-core/src/common/permissionInterface.ts:14]: You mean whether the account has the permission? Then sth. like NoPermission may be better?

Comment (SLdragon) [reply packages/fx-core/src/common/permissionInterface.ts:14]: it is not just no permission, it is M365 account in different tenant

Comment (KennethBWSong) [reply packages/fx-core/src/common/permissionInterface.ts:14]: That's OK, and maybe we can consider about a better naming here.

Comment (adashen) [reply packages/fx-core/src/common/permissionInterface.ts:14]: Can we re-name it to show the tenant not match?

File: packages/fx-core/src/plugins/solution/fx-solution/solution.ts [hunk 2]
```diff
@@ -1512,21 +1543,24 @@ export class TeamsAppSolution implements Solution {
           : "undefined",
       });
 
-      return ok(permissions);
+      return ok({
+        state: CollaborationState.OK,
+        permissions,
+      });
     } finally {
       ctx.envInfo.profile.get(GLOBAL_CONFIG)?.delete(USER_INFO);
```
Comment (1yefuwang1) [inline packages/fx-core/src/plugins/solution/fx-solution/solution.ts:1551]: Seems like you are using the config interface as an message passing mechanism... which is not config interface was intended to be used.

Comment (SLdragon) [reply packages/fx-core/src/plugins/solution/fx-solution/solution.ts:1551]: Thanks, as discussed offline, we can add parameters to plugin to not use config. I will add a task for this

**Review by 1yefuwang1 — APPROVED** [review]

File: packages/fx-core/src/common/permissionInterface.ts (hunk 1 above)
Comment (SLdragon) [reply packages/fx-core/src/common/permissionInterface.ts:14]: Sure, will update

**Review by jayzhang — APPROVED** [review]

**Review by adashen — APPROVED** [review]

**Pull request merged**